From 02b37824512d79d854adf1ec59c399dc3d991312 Mon Sep 17 00:00:00 2001
From: ctang <ctang@cloudera.com>
Date: Thu, 25 Feb 2016 16:05:05 -0500
Subject: [PATCH 0520/1164] CDH-37515: HIVE-13082: Enable constant propagation
 optimization in query with left semi join (part3:
 fix the tests for mr1)

Change-Id: I454a9fd836e86b4766cedc5ae51a19b83cb025ec
---
 .../queries/clientpositive/constprog_semijoin.q    |   16 ++++----
 .../clientpositive/constprog_semijoin.q.out        |   42 ++++++++++----------
 .../clientpositive/spark/constprog_semijoin.q.out  |   32 +++++++--------
 .../clientpositive/tez/constprog_semijoin.q.out    |   42 ++++++++++----------
 4 files changed, 66 insertions(+), 66 deletions(-)

diff --git a/ql/src/test/queries/clientpositive/constprog_semijoin.q b/ql/src/test/queries/clientpositive/constprog_semijoin.q
index 4019616..b0c8ebf 100644
--- a/ql/src/test/queries/clientpositive/constprog_semijoin.q
+++ b/ql/src/test/queries/clientpositive/constprog_semijoin.q
@@ -10,25 +10,25 @@ create table table3 (id int);
 insert into table3 values (100), (100), (101), (102), (103);
 
 explain select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id where table1.val = 't1val01';
-select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id where table1.val = 't1val01';
+select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id where table1.val = 't1val01' order by table1.id, table1.val, table1.val1;
 
 explain select table1.id, table1.val, table2.val2 from table1 inner join table2 on table1.val = 't1val01' and table1.id = table2.id left semi join table3 on table1.dimid = table3.id;
-select table1.id, table1.val, table2.val2 from table1 inner join table2 on table1.val = 't1val01' and table1.id = table2.id left semi join table3 on table1.dimid = table3.id;
+select table1.id, table1.val, table2.val2 from table1 inner join table2 on table1.val = 't1val01' and table1.id = table2.id left semi join table3 on table1.dimid = table3.id order by table1.id, table1.val, table2.val2;
 
 explain select table1.id, table1.val, table2.val2 from table1 left semi join table3 on table1.dimid = table3.id inner join table2 on table1.val = 't1val01' and table1.id = table2.id;
-select table1.id, table1.val, table2.val2 from table1 left semi join table3 on table1.dimid = table3.id inner join table2 on table1.val = 't1val01' and table1.id = table2.id;
+select table1.id, table1.val, table2.val2 from table1 left semi join table3 on table1.dimid = table3.id inner join table2 on table1.val = 't1val01' and table1.id = table2.id order by table1.id, table1.val, table2.val2;
 
 explain select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid <> 100;
-select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid <> 100;
+select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid <> 100 order by table1.id, table1.val, table1.val1;
 
 explain select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  IN (100,200);
-select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  IN (100,200);
+select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  IN (100,200) order by table1.id, table1.val, table1.val1;
 
 explain select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 200;
-select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 200;
+select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 200 order by table1.id, table1.val, table1.val1;
 
 explain select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 100;
-select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 100;
+select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 100 order by table1.id, table1.val, table1.val1;
 
 explain select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100;
-select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100;
+select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 order by table1.id, table1.val, table1.val1;
diff --git a/ql/src/test/results/clientpositive/constprog_semijoin.q.out b/ql/src/test/results/clientpositive/constprog_semijoin.q.out
index 9ecf849..105f4c0 100644
--- a/ql/src/test/results/clientpositive/constprog_semijoin.q.out
+++ b/ql/src/test/results/clientpositive/constprog_semijoin.q.out
@@ -128,18 +128,18 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id where table1.val = 't1val01'
+PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id where table1.val = 't1val01' order by table1.id, table1.val, table1.val1
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id where table1.val = 't1val01'
+POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id where table1.val = 't1val01' order by table1.id, table1.val, table1.val1
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table3
 #### A masked pattern was here ####
-3	t1val01	val104
 1	t1val01	val101
+3	t1val01	val104
 3	t1val01	val106
 PREHOOK: query: explain select table1.id, table1.val, table2.val2 from table1 inner join table2 on table1.val = 't1val01' and table1.id = table2.id left semi join table3 on table1.dimid = table3.id
 PREHOOK: type: QUERY
@@ -251,21 +251,21 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table2.val2 from table1 inner join table2 on table1.val = 't1val01' and table1.id = table2.id left semi join table3 on table1.dimid = table3.id
+PREHOOK: query: select table1.id, table1.val, table2.val2 from table1 inner join table2 on table1.val = 't1val01' and table1.id = table2.id left semi join table3 on table1.dimid = table3.id order by table1.id, table1.val, table2.val2
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table2
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table2.val2 from table1 inner join table2 on table1.val = 't1val01' and table1.id = table2.id left semi join table3 on table1.dimid = table3.id
+POSTHOOK: query: select table1.id, table1.val, table2.val2 from table1 inner join table2 on table1.val = 't1val01' and table1.id = table2.id left semi join table3 on table1.dimid = table3.id order by table1.id, table1.val, table2.val2
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table2
 POSTHOOK: Input: default@table3
 #### A masked pattern was here ####
-3	t1val01	t2val203
 1	t1val01	t2val201
 3	t1val01	t2val203
+3	t1val01	t2val203
 PREHOOK: query: explain select table1.id, table1.val, table2.val2 from table1 left semi join table3 on table1.dimid = table3.id inner join table2 on table1.val = 't1val01' and table1.id = table2.id
 PREHOOK: type: QUERY
 POSTHOOK: query: explain select table1.id, table1.val, table2.val2 from table1 left semi join table3 on table1.dimid = table3.id inner join table2 on table1.val = 't1val01' and table1.id = table2.id
@@ -375,13 +375,13 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table2.val2 from table1 left semi join table3 on table1.dimid = table3.id inner join table2 on table1.val = 't1val01' and table1.id = table2.id
+PREHOOK: query: select table1.id, table1.val, table2.val2 from table1 left semi join table3 on table1.dimid = table3.id inner join table2 on table1.val = 't1val01' and table1.id = table2.id order by table1.id, table1.val, table2.val2
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table2
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table2.val2 from table1 left semi join table3 on table1.dimid = table3.id inner join table2 on table1.val = 't1val01' and table1.id = table2.id
+POSTHOOK: query: select table1.id, table1.val, table2.val2 from table1 left semi join table3 on table1.dimid = table3.id inner join table2 on table1.val = 't1val01' and table1.id = table2.id order by table1.id, table1.val, table2.val2
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table2
@@ -461,12 +461,12 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid <> 100
+PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid <> 100 order by table1.id, table1.val, table1.val1
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid <> 100
+POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid <> 100 order by table1.id, table1.val, table1.val1
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table3
@@ -542,18 +542,18 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  IN (100,200)
+PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  IN (100,200) order by table1.id, table1.val, table1.val1
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  IN (100,200)
+POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  IN (100,200) order by table1.id, table1.val, table1.val1
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table3
 #### A masked pattern was here ####
-3	t1val01	val104
 1	t1val01	val101
+3	t1val01	val104
 PREHOOK: query: explain select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 200
 PREHOOK: type: QUERY
 POSTHOOK: query: explain select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 200
@@ -625,12 +625,12 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 200
+PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 200 order by table1.id, table1.val, table1.val1
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 200
+POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 200 order by table1.id, table1.val, table1.val1
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table3
@@ -706,18 +706,18 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 100
+PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 100 order by table1.id, table1.val, table1.val1
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 100
+POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 100 order by table1.id, table1.val, table1.val1
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table3
 #### A masked pattern was here ####
-3	t1val01	val104
 1	t1val01	val101
+3	t1val01	val104
 PREHOOK: query: explain select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100
 PREHOOK: type: QUERY
 POSTHOOK: query: explain select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100
@@ -789,15 +789,15 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100
+PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 order by table1.id, table1.val, table1.val1
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100
+POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 order by table1.id, table1.val, table1.val1
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table3
 #### A masked pattern was here ####
-3	t1val01	val104
 1	t1val01	val101
+3	t1val01	val104
diff --git a/ql/src/test/results/clientpositive/spark/constprog_semijoin.q.out b/ql/src/test/results/clientpositive/spark/constprog_semijoin.q.out
index a1536ae..6fec254 100644
--- a/ql/src/test/results/clientpositive/spark/constprog_semijoin.q.out
+++ b/ql/src/test/results/clientpositive/spark/constprog_semijoin.q.out
@@ -132,12 +132,12 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id where table1.val = 't1val01'
+PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id where table1.val = 't1val01' order by table1.id, table1.val, table1.val1
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id where table1.val = 't1val01'
+POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id where table1.val = 't1val01' order by table1.id, table1.val, table1.val1
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table3
@@ -255,13 +255,13 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table2.val2 from table1 inner join table2 on table1.val = 't1val01' and table1.id = table2.id left semi join table3 on table1.dimid = table3.id
+PREHOOK: query: select table1.id, table1.val, table2.val2 from table1 inner join table2 on table1.val = 't1val01' and table1.id = table2.id left semi join table3 on table1.dimid = table3.id order by table1.id, table1.val, table2.val2
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table2
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table2.val2 from table1 inner join table2 on table1.val = 't1val01' and table1.id = table2.id left semi join table3 on table1.dimid = table3.id
+POSTHOOK: query: select table1.id, table1.val, table2.val2 from table1 inner join table2 on table1.val = 't1val01' and table1.id = table2.id left semi join table3 on table1.dimid = table3.id order by table1.id, table1.val, table2.val2
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table2
@@ -379,13 +379,13 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table2.val2 from table1 left semi join table3 on table1.dimid = table3.id inner join table2 on table1.val = 't1val01' and table1.id = table2.id
+PREHOOK: query: select table1.id, table1.val, table2.val2 from table1 left semi join table3 on table1.dimid = table3.id inner join table2 on table1.val = 't1val01' and table1.id = table2.id order by table1.id, table1.val, table2.val2
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table2
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table2.val2 from table1 left semi join table3 on table1.dimid = table3.id inner join table2 on table1.val = 't1val01' and table1.id = table2.id
+POSTHOOK: query: select table1.id, table1.val, table2.val2 from table1 left semi join table3 on table1.dimid = table3.id inner join table2 on table1.val = 't1val01' and table1.id = table2.id order by table1.id, table1.val, table2.val2
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table2
@@ -469,12 +469,12 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid <> 100
+PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid <> 100 order by table1.id, table1.val, table1.val1
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid <> 100
+POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid <> 100 order by table1.id, table1.val, table1.val1
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table3
@@ -554,12 +554,12 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  IN (100,200)
+PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  IN (100,200) order by table1.id, table1.val, table1.val1
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  IN (100,200)
+POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  IN (100,200) order by table1.id, table1.val, table1.val1
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table3
@@ -641,12 +641,12 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 200
+PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 200 order by table1.id, table1.val, table1.val1
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 200
+POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 200 order by table1.id, table1.val, table1.val1
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table3
@@ -726,12 +726,12 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 100
+PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 100 order by table1.id, table1.val, table1.val1
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 100
+POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 100 order by table1.id, table1.val, table1.val1
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table3
@@ -813,12 +813,12 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100
+PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 order by table1.id, table1.val, table1.val1
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100
+POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 order by table1.id, table1.val, table1.val1
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table3
diff --git a/ql/src/test/results/clientpositive/tez/constprog_semijoin.q.out b/ql/src/test/results/clientpositive/tez/constprog_semijoin.q.out
index 15d3acc..b538cf2 100644
--- a/ql/src/test/results/clientpositive/tez/constprog_semijoin.q.out
+++ b/ql/src/test/results/clientpositive/tez/constprog_semijoin.q.out
@@ -136,18 +136,18 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id where table1.val = 't1val01'
+PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id where table1.val = 't1val01' order by table1.id, table1.val, table1.val1
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id where table1.val = 't1val01'
+POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id where table1.val = 't1val01' order by table1.id, table1.val, table1.val1
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table3
 #### A masked pattern was here ####
-3	t1val01	val104
 1	t1val01	val101
+3	t1val01	val104
 3	t1val01	val106
 PREHOOK: query: explain select table1.id, table1.val, table2.val2 from table1 inner join table2 on table1.val = 't1val01' and table1.id = table2.id left semi join table3 on table1.dimid = table3.id
 PREHOOK: type: QUERY
@@ -259,21 +259,21 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table2.val2 from table1 inner join table2 on table1.val = 't1val01' and table1.id = table2.id left semi join table3 on table1.dimid = table3.id
+PREHOOK: query: select table1.id, table1.val, table2.val2 from table1 inner join table2 on table1.val = 't1val01' and table1.id = table2.id left semi join table3 on table1.dimid = table3.id order by table1.id, table1.val, table2.val2
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table2
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table2.val2 from table1 inner join table2 on table1.val = 't1val01' and table1.id = table2.id left semi join table3 on table1.dimid = table3.id
+POSTHOOK: query: select table1.id, table1.val, table2.val2 from table1 inner join table2 on table1.val = 't1val01' and table1.id = table2.id left semi join table3 on table1.dimid = table3.id order by table1.id, table1.val, table2.val2
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table2
 POSTHOOK: Input: default@table3
 #### A masked pattern was here ####
-3	t1val01	t2val203
 1	t1val01	t2val201
 3	t1val01	t2val203
+3	t1val01	t2val203
 PREHOOK: query: explain select table1.id, table1.val, table2.val2 from table1 left semi join table3 on table1.dimid = table3.id inner join table2 on table1.val = 't1val01' and table1.id = table2.id
 PREHOOK: type: QUERY
 POSTHOOK: query: explain select table1.id, table1.val, table2.val2 from table1 left semi join table3 on table1.dimid = table3.id inner join table2 on table1.val = 't1val01' and table1.id = table2.id
@@ -383,13 +383,13 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table2.val2 from table1 left semi join table3 on table1.dimid = table3.id inner join table2 on table1.val = 't1val01' and table1.id = table2.id
+PREHOOK: query: select table1.id, table1.val, table2.val2 from table1 left semi join table3 on table1.dimid = table3.id inner join table2 on table1.val = 't1val01' and table1.id = table2.id order by table1.id, table1.val, table2.val2
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table2
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table2.val2 from table1 left semi join table3 on table1.dimid = table3.id inner join table2 on table1.val = 't1val01' and table1.id = table2.id
+POSTHOOK: query: select table1.id, table1.val, table2.val2 from table1 left semi join table3 on table1.dimid = table3.id inner join table2 on table1.val = 't1val01' and table1.id = table2.id order by table1.id, table1.val, table2.val2
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table2
@@ -477,12 +477,12 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid <> 100
+PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid <> 100 order by table1.id, table1.val, table1.val1
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid <> 100
+POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid <> 100 order by table1.id, table1.val, table1.val1
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table3
@@ -566,18 +566,18 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  IN (100,200)
+PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  IN (100,200) order by table1.id, table1.val, table1.val1
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  IN (100,200)
+POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  IN (100,200) order by table1.id, table1.val, table1.val1
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table3
 #### A masked pattern was here ####
-3	t1val01	val104
 1	t1val01	val101
+3	t1val01	val104
 PREHOOK: query: explain select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 200
 PREHOOK: type: QUERY
 POSTHOOK: query: explain select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 200
@@ -657,12 +657,12 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 200
+PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 200 order by table1.id, table1.val, table1.val1
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 200
+POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 200 order by table1.id, table1.val, table1.val1
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table3
@@ -746,18 +746,18 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 100
+PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 100 order by table1.id, table1.val, table1.val1
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 100
+POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 where table1.dimid  = 100 order by table1.id, table1.val, table1.val1
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table3
 #### A masked pattern was here ####
-3	t1val01	val104
 1	t1val01	val101
+3	t1val01	val104
 PREHOOK: query: explain select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100
 PREHOOK: type: QUERY
 POSTHOOK: query: explain select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100
@@ -837,15 +837,15 @@ STAGE PLANS:
       Processor Tree:
         ListSink
 
-PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100
+PREHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 order by table1.id, table1.val, table1.val1
 PREHOOK: type: QUERY
 PREHOOK: Input: default@table1
 PREHOOK: Input: default@table3
 #### A masked pattern was here ####
-POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100
+POSTHOOK: query: select table1.id, table1.val, table1.val1 from table1 left semi join table3 on table1.dimid = table3.id and table3.id = 100 order by table1.id, table1.val, table1.val1
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@table1
 POSTHOOK: Input: default@table3
 #### A masked pattern was here ####
-3	t1val01	val104
 1	t1val01	val101
+3	t1val01	val104
-- 
1.7.9.5

